storage/store.go

```diff
@@ -1452,9 +1452,51 @@ func (s *store) canUseShifting(uidmap, gidmap []idtools.IDMap) bool {
 // On entry:
 // - rlstore must be locked for writing
 // - rlstores MUST NOT be locked
-func (s *store) putLayer(rlstore rwLayerStore, rlstores []roLayerStore, id, parent string, names []string, mountLabel string, writeable bool, lOptions *LayerOptions, diff io.Reader, slo *stagedLayerOptions) (*Layer, int64, error) {
+func (s *store) putLayer(rlstore rwLayerStore, id string, parentLayer *Layer, names []string, mountLabel string, writeable bool, options *LayerOptions, lc *layerCreationContents) (*Layer, int64, error) {
+	return rlstore.create(id, parentLayer, names, mountLabel, nil, options, writeable, lc)
+}
+
+// On entry:
+// - rlstore must be locked for reading or writing
+// - rlstores MUST NOT be locked
+// Returns an extra unlock function to unlock any potentially read locked rlstores by this function.
+// The unlock function is always set and thus must always be called.
+func getParentLayer(rlstore roLayerStore, rlstores []roLayerStore, parent string) (*Layer, func(), error) {
+	// function we return to the caller so the caller gets the right stores locked and can unlock at the proper time themselves
+	var lockedLayerStores []roLayerStore
+	unlock := func() {
+		for _, i := range lockedLayerStores {
+			i.stopReading()
+		}
+	}
+	for _, l := range append([]roLayerStore{rlstore}, rlstores...) {
+		lstore := l
+		if lstore != rlstore {
+			if err := lstore.startReading(); err != nil {
+				return nil, unlock, err
+			}
+			lockedLayerStores = append(lockedLayerStores, lstore)
+		}
+		if l, err := lstore.Get(parent); err == nil && l != nil {
+			return l, unlock, nil
+		}
+	}
+
+	return nil, unlock, ErrLayerUnknown
+}
+
+// On entry:
+// - rlstore must be locked for reading or writing
+// - rlstores MUST NOT be locked
+//
+// Returns the new copied LayerOptions with mappings set, the parent Layer and
+// an extra unlock function to unlock any potentially read locked rlstores by this function.
+// The unlock function is always set and thus must always be called.
+func populateLayerOptions(s *store, rlstore roLayerStore, rlstores []roLayerStore, parent string, lOptions *LayerOptions) (*LayerOptions, *Layer, func(), error) {
```

Comment:
At least
```suggestion
func populateLayerOptions(s *store, rlstore roLayerStore, rlstores []roLayerStore, parent string, lOptions *LayerOptions) (*LayerOptions, *Layer, func(), error) {
  // WARNING: Update also the freshLayer checks in store.PutLayer if adding more logic here.
```
please, for https://github.com/containers/container-libs/pull/378#discussion_r2528956792 .